// @smithery/registry/dist/commonjs/models/operations/listservers.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListServersResponse$ = exports.ListServersResponse$outboundSchema = exports.ListServersResponse$inboundSchema = exports.ListServersRequest$ = exports.ListServersRequest$outboundSchema = exports.ListServersRequest$inboundSchema = void 0;
exports.listServersRequestToJSON = listServersRequestToJSON;
exports.listServersRequestFromJSON = listServersRequestFromJSON;
exports.listServersResponseToJSON = listServersResponseToJSON;
exports.listServersResponseFromJSON = listServersResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.ListServersRequest$inboundSchema = z.object({
    q: z.string().optional(),
    page: z.number().int().default(1),
    pageSize: z.number().int().default(10),
});
/** @internal */
exports.ListServersRequest$outboundSchema = z.object({
    q: z.string().optional(),
    page: z.number().int().default(1),
    pageSize: z.number().int().default(10),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListServersRequest$;
(function (ListServersRequest$) {
    /** @deprecated use `ListServersRequest$inboundSchema` instead. */
    ListServersRequest$.inboundSchema = exports.ListServersRequest$inboundSchema;
    /** @deprecated use `ListServersRequest$outboundSchema` instead. */
    ListServersRequest$.outboundSchema = exports.ListServersRequest$outboundSchema;
})(ListServersRequest$ || (exports.ListServersRequest$ = ListServersRequest$ = {}));
function listServersRequestToJSON(listServersRequest) {
    return JSON.stringify(exports.ListServersRequest$outboundSchema.parse(listServersRequest));
}
function listServersRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListServersRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListServersRequest' from JSON`);
}
/** @internal */
exports.ListServersResponse$inboundSchema = z.object({
    Result: components.ServerListResponse$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.ListServersResponse$outboundSchema = z.object({
    result: components.ServerListResponse$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListServersResponse$;
(function (ListServersResponse$) {
    /** @deprecated use `ListServersResponse$inboundSchema` instead. */
    ListServersResponse$.inboundSchema = exports.ListServersResponse$inboundSchema;
    /** @deprecated use `ListServersResponse$outboundSchema` instead. */
    ListServersResponse$.outboundSchema = exports.ListServersResponse$outboundSchema;
})(ListServersResponse$ || (exports.ListServersResponse$ = ListServersResponse$ = {}));
function listServersResponseToJSON(listServersResponse) {
    return JSON.stringify(exports.ListServersResponse$outboundSchema.parse(listServersResponse));
}
function listServersResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListServersResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListServersResponse' from JSON`);
}
