// @smithery/registry/dist/commonjs/models/components/connectioninfo.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionInfo$ = exports.ConnectionInfo$outboundSchema = exports.ConnectionInfo$inboundSchema = exports.ConnectionInfoType$ = exports.ConnectionInfoType$outboundSchema = exports.ConnectionInfoType$inboundSchema = exports.ConnectionInfoType = void 0;
exports.connectionInfoToJSON = connectionInfoToJSON;
exports.connectionInfoFromJSON = connectionInfoFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Connection type
 */
exports.ConnectionInfoType = {
    Http: "http",
    Stdio: "stdio",
};
/** @internal */
exports.ConnectionInfoType$inboundSchema = z.nativeEnum(exports.ConnectionInfoType);
/** @internal */
exports.ConnectionInfoType$outboundSchema = exports.ConnectionInfoType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConnectionInfoType$;
(function (ConnectionInfoType$) {
    /** @deprecated use `ConnectionInfoType$inboundSchema` instead. */
    ConnectionInfoType$.inboundSchema = exports.ConnectionInfoType$inboundSchema;
    /** @deprecated use `ConnectionInfoType$outboundSchema` instead. */
    ConnectionInfoType$.outboundSchema = exports.ConnectionInfoType$outboundSchema;
})(ConnectionInfoType$ || (exports.ConnectionInfoType$ = ConnectionInfoType$ = {}));
/** @internal */
exports.ConnectionInfo$inboundSchema = z.object({
    type: exports.ConnectionInfoType$inboundSchema,
    deploymentUrl: z.string().optional(),
    configSchema: z.record(z.any()),
    published: z.boolean().optional(),
    stdioFunction: z.string().optional(),
});
/** @internal */
exports.ConnectionInfo$outboundSchema = z.object({
    type: exports.ConnectionInfoType$outboundSchema,
    deploymentUrl: z.string().optional(),
    configSchema: z.record(z.any()),
    published: z.boolean().optional(),
    stdioFunction: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConnectionInfo$;
(function (ConnectionInfo$) {
    /** @deprecated use `ConnectionInfo$inboundSchema` instead. */
    ConnectionInfo$.inboundSchema = exports.ConnectionInfo$inboundSchema;
    /** @deprecated use `ConnectionInfo$outboundSchema` instead. */
    ConnectionInfo$.outboundSchema = exports.ConnectionInfo$outboundSchema;
})(ConnectionInfo$ || (exports.ConnectionInfo$ = ConnectionInfo$ = {}));
function connectionInfoToJSON(connectionInfo) {
    return JSON.stringify(exports.ConnectionInfo$outboundSchema.parse(connectionInfo));
}
function connectionInfoFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ConnectionInfo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ConnectionInfo' from JSON`);
}
