// @smithery/registry/dist/commonjs/models/components/serverdetailresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerDetailResponse$ = exports.ServerDetailResponse$outboundSchema = exports.ServerDetailResponse$inboundSchema = exports.ServerDetailResponseSecurity$ = exports.ServerDetailResponseSecurity$outboundSchema = exports.ServerDetailResponseSecurity$inboundSchema = void 0;
exports.serverDetailResponseSecurityToJSON = serverDetailResponseSecurityToJSON;
exports.serverDetailResponseSecurityFromJSON = serverDetailResponseSecurityFromJSON;
exports.serverDetailResponseToJSON = serverDetailResponseToJSON;
exports.serverDetailResponseFromJSON = serverDetailResponseFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const connectioninfo_js_1 = require("./connectioninfo.js");
const tool_js_1 = require("./tool.js");
/** @internal */
exports.ServerDetailResponseSecurity$inboundSchema = z.object({
    scanPassed: z.nullable(z.boolean()).optional(),
});
/** @internal */
exports.ServerDetailResponseSecurity$outboundSchema = z.object({
    scanPassed: z.nullable(z.boolean()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServerDetailResponseSecurity$;
(function (ServerDetailResponseSecurity$) {
    /** @deprecated use `ServerDetailResponseSecurity$inboundSchema` instead. */
    ServerDetailResponseSecurity$.inboundSchema = exports.ServerDetailResponseSecurity$inboundSchema;
    /** @deprecated use `ServerDetailResponseSecurity$outboundSchema` instead. */
    ServerDetailResponseSecurity$.outboundSchema = exports.ServerDetailResponseSecurity$outboundSchema;
})(ServerDetailResponseSecurity$ || (exports.ServerDetailResponseSecurity$ = ServerDetailResponseSecurity$ = {}));
function serverDetailResponseSecurityToJSON(serverDetailResponseSecurity) {
    return JSON.stringify(exports.ServerDetailResponseSecurity$outboundSchema.parse(serverDetailResponseSecurity));
}
function serverDetailResponseSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ServerDetailResponseSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ServerDetailResponseSecurity' from JSON`);
}
/** @internal */
exports.ServerDetailResponse$inboundSchema = z.object({
    qualifiedName: z.string(),
    displayName: z.string(),
    iconUrl: z.nullable(z.string()).optional(),
    remote: z.boolean().optional(),
    connections: z.array(connectioninfo_js_1.ConnectionInfo$inboundSchema),
    security: z.nullable(z.lazy(() => exports.ServerDetailResponseSecurity$inboundSchema))
        .optional(),
    tools: z.nullable(z.array(tool_js_1.Tool$inboundSchema)).optional(),
});
/** @internal */
exports.ServerDetailResponse$outboundSchema = z.object({
    qualifiedName: z.string(),
    displayName: z.string(),
    iconUrl: z.nullable(z.string()).optional(),
    remote: z.boolean().optional(),
    connections: z.array(connectioninfo_js_1.ConnectionInfo$outboundSchema),
    security: z.nullable(z.lazy(() => exports.ServerDetailResponseSecurity$outboundSchema)).optional(),
    tools: z.nullable(z.array(tool_js_1.Tool$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServerDetailResponse$;
(function (ServerDetailResponse$) {
    /** @deprecated use `ServerDetailResponse$inboundSchema` instead. */
    ServerDetailResponse$.inboundSchema = exports.ServerDetailResponse$inboundSchema;
    /** @deprecated use `ServerDetailResponse$outboundSchema` instead. */
    ServerDetailResponse$.outboundSchema = exports.ServerDetailResponse$outboundSchema;
})(ServerDetailResponse$ || (exports.ServerDetailResponse$ = ServerDetailResponse$ = {}));
function serverDetailResponseToJSON(serverDetailResponse) {
    return JSON.stringify(exports.ServerDetailResponse$outboundSchema.parse(serverDetailResponse));
}
function serverDetailResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ServerDetailResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ServerDetailResponse' from JSON`);
}
