// @smithery/registry/dist/commonjs/models/components/tool.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tool$ = exports.Tool$outboundSchema = exports.Tool$inboundSchema = exports.InputSchema$ = exports.InputSchema$outboundSchema = exports.InputSchema$inboundSchema = exports.ToolType$ = exports.ToolType$outboundSchema = exports.ToolType$inboundSchema = exports.ToolType = void 0;
exports.inputSchemaToJSON = inputSchemaToJSON;
exports.inputSchemaFromJSON = inputSchemaFromJSON;
exports.toolToJSON = toolToJSON;
exports.toolFromJSON = toolFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
exports.ToolType = {
    Object: "object",
};
/** @internal */
exports.ToolType$inboundSchema = z
    .nativeEnum(exports.ToolType);
/** @internal */
exports.ToolType$outboundSchema = exports.ToolType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ToolType$;
(function (ToolType$) {
    /** @deprecated use `ToolType$inboundSchema` instead. */
    ToolType$.inboundSchema = exports.ToolType$inboundSchema;
    /** @deprecated use `ToolType$outboundSchema` instead. */
    ToolType$.outboundSchema = exports.ToolType$outboundSchema;
})(ToolType$ || (exports.ToolType$ = ToolType$ = {}));
/** @internal */
exports.InputSchema$inboundSchema = (0, schemas_js_1.collectExtraKeys)(z.object({
    type: exports.ToolType$inboundSchema,
    properties: z.record(z.any()).optional(),
}).catchall(z.any()), "additionalProperties", true);
/** @internal */
exports.InputSchema$outboundSchema = z.object({
    type: exports.ToolType$outboundSchema,
    properties: z.record(z.any()).optional(),
    additionalProperties: z.record(z.any()),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...(0, primitives_js_1.remap)(v, {
            additionalProperties: null,
        }),
    };
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InputSchema$;
(function (InputSchema$) {
    /** @deprecated use `InputSchema$inboundSchema` instead. */
    InputSchema$.inboundSchema = exports.InputSchema$inboundSchema;
    /** @deprecated use `InputSchema$outboundSchema` instead. */
    InputSchema$.outboundSchema = exports.InputSchema$outboundSchema;
})(InputSchema$ || (exports.InputSchema$ = InputSchema$ = {}));
function inputSchemaToJSON(inputSchema) {
    return JSON.stringify(exports.InputSchema$outboundSchema.parse(inputSchema));
}
function inputSchemaFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.InputSchema$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InputSchema' from JSON`);
}
/** @internal */
exports.Tool$inboundSchema = z
    .object({
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    inputSchema: z.lazy(() => exports.InputSchema$inboundSchema),
});
/** @internal */
exports.Tool$outboundSchema = z.object({
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    inputSchema: z.lazy(() => exports.InputSchema$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Tool$;
(function (Tool$) {
    /** @deprecated use `Tool$inboundSchema` instead. */
    Tool$.inboundSchema = exports.Tool$inboundSchema;
    /** @deprecated use `Tool$outboundSchema` instead. */
    Tool$.outboundSchema = exports.Tool$outboundSchema;
})(Tool$ || (exports.Tool$ = Tool$ = {}));
function toolToJSON(tool) {
    return JSON.stringify(exports.Tool$outboundSchema.parse(tool));
}
function toolFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Tool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Tool' from JSON`);
}
